x-pack/legacy/plugins/infra/server/lib/snapshot/snapshot.ts

```diff
@@ -63,6 +69,24 @@ export class InfraSnapshot {
   }
 }
 
+// This is used by the getAllCompositeData to select the bucket to return.
+const bucketSelector = (
+  response: InfraDatabaseSearchResponse<{}, InfraSnapshotAggregationResponse>
+) => (response.aggregations && response.aggregations.nodes.buckets) || [];
+
+// This is used by getAllCompositeData to add the after key to the subsequent requests
+const handleAfterKey = (
```

Comment:
As the two uses of `handleAfterKey` in this PR are very, very similar - with only the actual keys and portions of the response to be used changing - is it worth making this into a helper function too? I imagine the 

```
  if (!response.aggregations) {
    return options;
  }
```

check and the merging of the `options` will always be the same.